control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go

```diff
@@ -919,3 +982,70 @@ func TestReconcileClusterVersion(t *testing.T) {
 	g.Expect(clusterVersion.Spec.Overrides).To(Equal(testOverrides))
 	g.Expect(clusterVersion.Spec.Channel).To(BeEmpty())
 }
+
+func TestReconcileImageContentPolicyType(t *testing.T) {
+	testCases := []struct {
+		name           string
+		hcp            *hyperv1.HostedControlPlane
+		expectedIDMS   *configv1.ImageDigestMirrorSet
+		expectedToFail bool
+		reconcileTwice bool
+	}{
+		{
+			name:         "ICS with content, it should return an IDMS with the same content",
+			hcp:          withICS(fakeHCP()),
+			expectedIDMS: fakeIDMS(),
+		},
+		{
+			name:           "ICS empty, is should return a not found error",
+			hcp:            fakeHCP(),
+			expectedToFail: true,
+		},
+		{
+			name:           "ICS with content and updated the HCP deleting the ICS, it should return a not found error",
+			hcp:            withICS(fakeHCP()),
+			expectedIDMS:   nil,
+			expectedToFail: true,
+			reconcileTwice: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			fakeClient := fake.NewClientBuilder().WithScheme(api.Scheme).WithObjects(tc.hcp).Build()
+			r := &reconciler{
+				client:                 fakeClient,
+				CreateOrUpdateProvider: &simpleCreateOrUpdater{},
+			}
+			err := r.reconcileImageContentPolicyType(context.Background(), tc.hcp)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			idms := globalconfig.ImageDigestMirrorSet()
+			err = fakeClient.Get(context.Background(), client.ObjectKeyFromObject(idms), idms)
+			if tc.hcp.Spec.ImageContentSources == nil && tc.expectedToFail {
+				g.Expect(errors.IsNotFound(err)).To(BeTrue(), "expecting a not-found error, received: %v", err)
+			}
+
+			if !tc.reconcileTwice && !tc.expectedToFail {
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(len(tc.hcp.Spec.ImageContentSources)).To(Equal(len(idms.Spec.ImageDigestMirrors)))
+				g.Expect(tc.expectedIDMS.Spec).To(BeEquivalentTo(idms.Spec))
+			}
+
+			if tc.reconcileTwice {
```

Comment:
can we change the name from reconcileTwice to something more meaningful like removeICSAndReconcile